libraries/Microsoft.Bot.Builder/LogTelemetryClient.cs

```diff
@@ -0,0 +1,94 @@
+        /// Examples are stored procedure name and URL path template.</param>
+        /// <param name="data">Command initiated by this dependency call. Examples are SQL statement and HTTP
+        /// URL's with all query parameters.</param>
+        /// <param name="startTime">The time when the dependency was called.</param>
+        /// <param name="duration">The time taken by the external dependency to handle the call.</param>
+        /// <param name="resultCode">Result code of dependency call execution.</param>
+        /// <param name="success">True if the dependency call was handled successfully.</param>
+        public virtual void TrackDependency(string dependencyTypeName, string target, string dependencyName, string data, DateTimeOffset startTime, TimeSpan duration, string resultCode, bool success)
+        {
+        }
+
+        /// <summary>
+        /// Logs custom events with extensible named fields.
+        /// </summary>
+        /// <param name="eventName">A name for the event.</param>
+        /// <param name="properties">Named string values you can use to search and classify events.</param>
+        /// <param name="metrics">Measurements associated with this event.</param>
+        public virtual void TrackEvent(string eventName, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
+        {
+        }
+
+        /// <summary>
+        /// Logs a system exception.
+        /// </summary>
+        /// <param name="exception">The exception to log.</param>
+        /// <param name="properties">Named string values you can use to classify and search for this exception.</param>
+        /// <param name="metrics">Additional values associated with this exception.</param>
+        public virtual void TrackException(Exception exception, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
+        {
+        }
+
+        /// <summary>
+        /// Send a trace message.
+        /// </summary>
+        /// <param name="message">Message to display.</param>
+        /// <param name="severityLevel">Trace severity level <see cref="Severity"/>.</param>
+        /// <param name="properties">Named string values you can use to search and classify events.</param>
+        public virtual void TrackTrace(string message, Severity severityLevel, IDictionary<string, string> properties)
+        {
+        }
+
+        /// <summary>
+        /// Logs a page view.
+        /// </summary>
+        /// <param name="name">The name of the page view to log.</param>
+        /// <param name="properties">Named string values you can use to search and classify events.</param>
+        /// <param name="metrics">Measurements associated with this event.</param>
+        public virtual void TrackPageView(string name, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
```

Comment:
nit: naming - TrackPageView is leaking an AppInsights specific concept to the top level LogTelemetryClient, which doesn't seem right. It'd be great if this corresponded with a specific BF related user interaction.

Reply:
Yeah, I went back a forth on this - I originally had it named TrackDialogView.  I think you are right - I'll probably go back.